src/main/VideoPlayers.tsx

```diff
@@ -0,0 +1,384 @@
+              video!.textTracks[trackIdx].mode = 'showing'; // thanks Firefox
+              } else {
+                textTrack.track.mode = 'hidden';
+              video!.textTracks[trackIdx].mode = 'hidden'; // thanks Firefox
+              }
+            }
+          });
+          const video = document.querySelector('video');
+          if (video) {
+            video.appendChild(track);
+          }
+        });
+      }
+    }
+
+    // External functions
+    useImperativeHandle(forwardRefThing, () => ({
+    // Renders the current frame in the video element to a canvas
+    // Returns the data url
+      captureVideo() {
+        const video = ref.current?.getInternalPlayer() as HTMLVideoElement
+        const canvas = document.createElement("canvas");
+        canvas.width = video.videoWidth;
+        canvas.height = video.videoHeight;
+        const canvasContext = canvas.getContext("2d");
+        if (canvasContext !== null) {
+          canvasContext.drawImage(video, 0, 0);
+          return canvas.toDataURL('image/png')
+        }
+      }
+    }));
+
+    const errorBoxStyle = css({
+      ...(!errorState) && {display: "none"},
+      borderColor: `${theme.error}`,
+      borderStyle: 'dashed',
+      fontWeight: 'bold',
+      padding: '10px',
+    })
+
+    const reactPlayerStyle = css({
+      aspectRatio: '16 / 9',    // Hard-coded for now because there are problems with updating this value at runtime
+
+      overflow: 'hidden', // Required for borderRadius to show
+      ...(first) && {borderTopLeftRadius: '5px'},
+      ...(first) && {borderBottomLeftRadius: '5px'},
+      ...(last) && {borderTopRightRadius: '5px'},
+      ...(last) && {borderBottomRightRadius: '5px'},
+    })
+
+    const render = () => {
+      if (!errorState) {
+        return (
+          <ReactPlayer url={url}
+            css={[backgroundBoxStyle(theme), reactPlayerStyle]}
+            ref={ref}
+            width="unset"
+            height="unset"
+            playing={isPlaying}
+            muted={!isPrimary}
+            onProgress={onProgressCallback}
+            progressInterval={100}
+            onReady={onReadyCallback}
+            onPlay={onPlay}
+            onEnded={onEndedCallback}
+            onError={onErrorCallback}
+            tabIndex={-1}
+            config={playerConfig}
+            disablePictureInPicture
+          />
+        );
+      } else {
+        return (
+          <div css={errorBoxStyle} role="alert">
+            <span>{t("video.loadError-text")} </span>
+          </div>
+        );
+      }
+    }
+
+    return (
+      <>
+        {render()}
+      </>
+    );
+
+  // return (
+  //   <div title="Video Player">
+  //     <video width="320" height="240" controls ref={vidRef}>
+  //     <source src="https://media.geeksforgeeks.org/wp-content/uploads/20190616234019/Canvas.move_.mp4" type="video/mp4" />
+  //     Your browser does not support the video tag.
+  //     </video>
+  //   </div>
+  // );
```

Comment:
Another one, though as with the others, maybe you have a reason for keeping this around.